### PR TITLE
Fixes #19261: Migration script for rudder.community.port

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/19261

### CODE REVIEW

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 1]
```diff
@@ -379,6 +380,21 @@ upgrade_masterfiles() {
...
+upgrade_conf_file() {
+
+  STEP="Upgrade /opt/rudder/etc/rudder-web.properties"
+  CONF_FILE="/opt/rudder/etc/rudder-web.properties"
+
+  # - 7.0.0~alpha1 : Rename property rudder.community.port into rudder.policy.distribution.port.cfengine
+  OLD_PROPERTY_NAME="rudder.community.port"
+  NEW_PROPERTY_NAME="rudder.policy.distribution.port.cfengine"
+  ATTRIBUTESET=$(grep -c "^${NEW_PROPERTY_NAME}[ \t]*=" ${CONF_FILE} || true)
+  if [ ${ATTRIBUTESET} -eq 0 ]; then
```
Comment (peckpeck) [inline rudder-webapp/SOURCES/rudder-upgrade:392]: I think his should be an "if old_property is present" instead

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 2]
```diff
@@ -379,6 +380,21 @@ upgrade_masterfiles() {
...
+    sed -e "s/$PROPERTY_NAME/rudder.policy.distribution.port.cfengine/" ${CONF_FILE} | grep -C 2 "port.cfengine"
```
Comment (peckpeck) [inline rudder-webapp/SOURCES/rudder-upgrade:393]: I don't understand this line

Comment (fanf) [reply rudder-webapp/SOURCES/rudder-upgrade:393]: That's... The test line, not the one that should be here.

File: rudder-webapp/SOURCES/rudder-upgrade (hunk 1 above)
Comment (fanf) [reply rudder-webapp/SOURCES/rudder-upgrade:392]: No, if the both are here (for ex, someone has added the new one in place or replacing the old one), then we don't want to do the sed, else we could have a duplicated property (and I'm not sure what happens in that case. First value taken? Last one?)

**Comment by fanf** [comment]
PR updated with a new commit

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 3]
```diff
@@ -379,6 +380,21 @@ upgrade_masterfiles() {
...
+
+  STEP="Upgrade /opt/rudder/etc/rudder-web.properties"
+  CONF_FILE="/opt/rudder/etc/rudder-web.properties"
+
+  # - 7.0.0~alpha1 : Rename property rudder.community.port into rudder.policy.distribution.port.cfengine
+  OLD_PROPERTY_NAME="rudder.community.port"
+  NEW_PROPERTY_NAME="rudder.policy.distribution.port.cfengine"
+  ATTRIBUTESET=$(grep -c "^${NEW_PROPERTY_NAME}[ \t]*=" ${CONF_FILE} || true)
+  if [ ${ATTRIBUTESET} -eq 0 ]; then
+    sed -i -e "s/$PROPERTY_NAME/rudder.policy.distribution.port.cfengine/" ${CONF_FILE}
```
Comment (peckpeck) [inline rudder-webapp/SOURCES/rudder-upgrade:393]: $PROPERTY_NAME is not defined

Comment (peckpeck) [reply rudder-webapp/SOURCES/rudder-upgrade:393]: if there are 2 entries (an old and a new), now there will be 2 conflciting lines

**Comment by fanf** [comment]
PR updated with a new commit

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 4]
```diff
@@ -379,6 +380,21 @@ upgrade_masterfiles() {
...
+
+  STEP="Upgrade /opt/rudder/etc/rudder-web.properties"
+  CONF_FILE="/opt/rudder/etc/rudder-web.properties"
+
+  # - 7.0.0~alpha1 : Rename property rudder.community.port into rudder.policy.distribution.port.cfengine
+  OLD_PROPERTY_NAME="rudder.community.port"
+  NEW_PROPERTY_NAME="rudder.policy.distribution.port.cfengine"
+  ATTRIBUTESET=$(grep -c "^${NEW_PROPERTY_NAME}[ \t]*=" ${CONF_FILE} || true)
+  if [ ${ATTRIBUTESET} -eq 0 ]; then
+    sed -i -e "s/${OLD_PROPERTY_NAME}/${NEW_PROPERTY_NAME}/" ${CONF_FILE}
```
Comment (peckpeck) [inline rudder-webapp/SOURCES/rudder-upgrade:393]: this will still leave 2 configuration line if there is one of each at the beginning

Comment (fanf) [reply rudder-webapp/SOURCES/rudder-upgrade:393]: yes, but that's not a problem since we take the new one first

Comment (peckpeck) [reply rudder-webapp/SOURCES/rudder-upgrade:393]: how do you know it's the new one ? they both have the same name after this

Comment (fanf) [reply rudder-webapp/SOURCES/rudder-upgrade:393]: I don't understand ?
If the user added `rudder.policy.distribution.port.cfengine`, then we have both `rudder.policy.distribution.port.cfengine` and `rudder.community.port` but the `if` prevent the sed; rudder takes `rudder.policy.distribution.port.cfengine` as expected. 
If the user didn't do anything (nominal case), then there is only `rudder.community.port`, the sed is done and rudder use the only value. 
If somehow the upgrade script doesn't run, then doesn't find `rudder.policy.distribution.port.cfengine`, so it fall back to `rudder.community.port`.

Comment (peckpeck) [reply rudder-webapp/SOURCES/rudder-upgrade:393]: aah i understood the if the other way around

**Review by peckpeck — APPROVED** [review]

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**